app/scripts/metamask-controller.js

```diff
@@ -6805,10 +6805,13 @@ export default class MetamaskController extends EventEmitter {
         listAccounts: this.accountsController.listAccounts.bind(
           this.accountsController,
         ),
-        requestPermissionsForOrigin: (requestedPermissions) =>
+        requestPermissionsForOrigin: (requestedPermissions, metadata) =>
           this.permissionController.requestPermissions(
             { origin },
             requestedPermissions,
+            {
+              metadata,
+            },
```

Comment:
does this not result in metadata.metadata at the UI level?

Reply:
nope since the third arg to requestPermissions is an "options" object that contains a metadata property